app/connection/ConnectionManager.js

```diff
@@ -122,7 +136,7 @@ class ConnectionManager extends EventEmitter {
         }
 
         this._websocket.send(JSON.stringify(req));
-        this._pendingRequests[req.req_id] = new PendingPromise(req);
+        this._pendingRequests[req.req_id] = promise ? promise: new PendingPromise(req);
```

Comment:
I think it's better to have `req_id` passed to `send` as an optional parameter. For this part, instead of using a passed promise, check if the pending request exist inside the object and reuse that instead